### PR TITLE
passing the test !

### PR DESCRIPTION
**Pull request by Leoli0320**
this version only passing the test with the vcr
haven't done with refactor, test coverage
please help me checking with the code

flay tweet_spec.rb  => Total score = 0

### CODE REVIEW

**Comment by cyhsutw** [comment]
I've added test coverage reporting feature in #10, could you merge `master` in first?

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,9 @@
 ruby '2.3.1'
 
 source 'https://rubygems.org'
-
+gem 'vcr'
```
Comment (cyhsutw) [inline Gemfile]: I would like the gems being ordered alphabetically

File: Gemfile [hunk 2]
```diff
@@ -12,3 +14,5 @@ gem 'pry-coolline'
 
 gem 'rake'
 gem 'rubocop'
+
+
```
Comment (cyhsutw) [inline Gemfile:18]: Please ✂️ these blank lines

File: config/script.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
```
Comment (cyhsutw) [inline config/script.rb]: Any specific reasons placing this file under `config`?

File: spec/twitter/coverage/.last_run.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (cyhsutw) [inline spec/twitter/coverage/.last_run.json]: Should be ignored by Git

File: spec/twitter/coverage/assets/0.10.0/application.css [hunk 5]
```diff
@@ -0,0 +1,799 @@
+/* -----------------------------------------------------------------------
```
Comment (cyhsutw) [inline spec/twitter/coverage/assets/0.10.0/application.css]: Ditto, ignore this

File: spec/twitter/test.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
```
Comment (cyhsutw) [inline spec/twitter/test.rb]: What's the purpose of this file, could you provide more details please 🙏 ?

File: spec/twitter/tweet_spec.rb [hunk 7]
```diff
@@ -1,13 +1,37 @@
 # frozen_string_literal: true
-require './spec/spec_helper.rb'
-require './lib/twitter/client.rb'
-require './lib/twitter/tweet.rb'
+require '../spec_helper.rb'
```
Comment (cyhsutw) [inline spec/twitter/tweet_spec.rb]: 👍🤘

**Review by cyhsutw — CHANGES_REQUESTED** [review]
I have re-write the code again, and commit it !

**Comment by cyhsutw** [comment]
@Leoli0320  you might need to resolve the conflicts first, or you can close this one and do it again on another branch then open a new pull request. 

It should be much more easier to open a new pull request since I've added `simplecov` on `master` and all you need to do is stubbing requests using  `vcr` and `webmock`.

File: config/script.rb (hunk 3 above)
Comment (Leoli0320) [reply config/script.rb]: Haha is my test XD

File: spec/twitter/coverage/.last_run.json (hunk 4 above)
Comment (Leoli0320) [reply spec/twitter/coverage/.last_run.json]: Ok

File: spec/twitter/coverage/assets/0.10.0/application.css (hunk 5 above)
Comment (Leoli0320) [reply spec/twitter/coverage/assets/0.10.0/application.css]: Ok

File: spec/twitter/test.rb (hunk 6 above)
Comment (Leoli0320) [reply spec/twitter/test.rb]: Haaaa only for my test , I will delete it. Sorry for bothering you.

**Review by cyhsutw — APPROVED** [review]

**Comment by cyhsutw** [comment]
@edsong1257 @Leoli0320 Just for the heads up, I updated the code:
1. The names of the cassettes are now automatically generated based on the spec description and the file name of the spec.
2. Remove unused files
3. Updated README.md

### Test results:

``` bash
$ bundle exec rake test
Run options: --seed 45136

# Running:

..

Finished in 0.079843s, 25.0491 runs/s, 25.0491 assertions/s.

2 runs, 2 assertions, 0 failures, 0 errors, 0 skips
Coverage report generated for RSpec to /Users/cyhsutw/Projects/temp/twitter-api/coverage. 34 / 34 LOC (100.0%) covered.
```

### Quality metrics

``` bash
$ bundle exec rake quality
bundle exec rubocop lib
Inspecting 2 files
..

2 files inspected, no offenses detected
bundle exec flay lib
Total score (lower is better) = 0
bundle exec flog lib
    23.7: flog total
     3.0: flog/method average

     7.3: Twitter::Client#search_tweets    lib/twitter/client.rb:15
     5.2: Twitter::Tweet#initialize        lib/twitter/tweet.rb:6
     3.6: Twitter::Tweet::search           lib/twitter/tweet.rb:12
```

If there's no further problems, I think it's ready to go. 
Please merge if you'd like to.

File: spec/spec_helper.rb [hunk 8]
```diff
@@ -1,7 +1,30 @@
...
 TAGS = ['#food', '#yum'].freeze
+
+VCR.configure do |config|
+  config.cassette_library_dir = CASSETTES_FOLDER
+  config.hook_into :webmock
+
+  config.filter_sensitive_data('<ACCESS_TOKEN>') { CREDENTIALS[:access_token] }
+end
+
+def cassette_name(path_to_spec, description)
```
Comment (cyhsutw) [inline spec/spec_helper.rb:25]: So here's the method that generates the names of the cassettes.

File: spec/twitter/tweet_spec.rb [hunk 9]
```diff
@@ -2,6 +2,18 @@
 require_relative '../spec_helper.rb'
 
 describe 'Twitter::Tweet::search' do
+VCR.configure do |c|
```
Comment (cyhsutw) [inline spec/twitter/tweet_spec.rb]: 這個在 `spec_helper.rb` 已經設定過一次了，應該不用再設定了

**Pull request merged**